File: VB/PrintingExample/MyPrintHelper.vb
#Region "#MyPrintHelper"
Imports DevExpress.Mvvm
Imports DevExpress.Xpf.Printing
Imports DevExpress.Xpf.Scheduling
Imports System.Windows

Namespace PrintingExample

    Public Module MyPrintHelper

        Public Property mainWindow As Window

        Public Sub PrintScheduler(ByVal scheduler As SchedulerControl)
            Dim report As XtraSchedulerReport1 = New XtraSchedulerReport1()
            Dim dateTimeRange As DateTimeRange = scheduler.VisibleIntervals(0)
            scheduler.SchedulerPrintAdapter.DateTimeRange = dateTimeRange
            scheduler.SchedulerPrintAdapter.AssignToReport(report)
            PrintHelper.ShowPrintPreview(mainWindow, report)
        End Sub
    End Module
End Namespace
#End Region  ' #MyPrintHelper

File: CS/PrintingExample/MyPrintHelper.cs
#region #MyPrintHelper
using DevExpress.Mvvm;
using DevExpress.Xpf.Printing;
using DevExpress.Xpf.Printing.Native;
using DevExpress.Xpf.Scheduling;
using DevExpress.XtraPrinting.Native;
using System.Windows;

namespace PrintingExample {
    public static class MyPrintHelper {
        public static Window mainWindow { get; set; }

        public static void PrintScheduler(SchedulerControl scheduler) {
            XtraSchedulerReport1 report = new XtraSchedulerReport1();
            DateTimeRange dateTimeRange = scheduler.VisibleIntervals[0];
            scheduler.SchedulerPrintAdapter.DateTimeRange = dateTimeRange;
            scheduler.SchedulerPrintAdapter.AssignToReport(report);
            PrintHelper.ShowPrintPreview(mainWindow, report);
        }
    }
}
#endregion #MyPrintHelper
